File: leetcode/javascript/5. Longest Palindromic Substring.js
const findLongestPalindrome = (s, i, j) => {
    while (i >= 0 && j < s.length && s[i] === s[j]) {
        i -= 1;
        j += 1;
    }

    return s.slice(i + 1, j);
}

var longestPalindrome = function(s) {
    let answer = '';

    for (let i = 0; i < s.length; i += 1) {
        const oddPalindrome = findLongestPalindrome(s, i, i);
        const evenPalindrome = findLongestPalindrome(s, i, i + 1);

        const ithLongestPalindrome = oddPalindrome.length > evenPalindrome.length ? oddPalindrome : evenPalindrome;

        answer = ithLongestPalindrome.length > answer.length ? ithLongestPalindrome : answer;
    }

    return answer;
};

File: leetcode/python/5. Longest Palindromic Substring.py
def findLongestPalindrome(s: str, i: int, j: int):
    while i >= 0 and j < len(s) and s[i] == s[j]:
        i -= 1
        j += 1

    return s[i + 1:j]

class Solution:
    def longestPalindrome(self, s: str) -> str:
        answer = ''

        for i in range(len(s)):
            oddPalindrome = findLongestPalindrome(s, i, i)
            evenPalindrome = findLongestPalindrome(s, i, i + 1)

            ithLongestPalindrome = oddPalindrome if len(oddPalindrome) > len(evenPalindrome) else evenPalindrome

            answer = ithLongestPalindrome if len(ithLongestPalindrome) > len(answer) else answer

        return answer